examples/js/loaders/OBJLoader.js

```diff
@@ -40,315 +65,431 @@ THREE.OBJLoader.prototype = {
+					);
 
-				// ["f 1/1 2/2 3/3", " 1/1", "1", "1", " 2/2", "2", "2", " 3/3", "3", "3", undefined, undefined, undefined]
+				} else if ( ( result = this.regexp.face_vertex_normal.exec( line ) ) !== null ) {
 
-				addFace(
-					result[ 2 ], result[ 5 ], result[ 8 ], result[ 11 ],
-					result[ 3 ], result[ 6 ], result[ 9 ], result[ 12 ]
-				);
+					// f vertex//normal vertex//normal vertex//normal
+					// 0                     1    2    3    4    5    6   7          8
+					// ["f 1//1 2//2 3//3", "1", "1", "2", "2", "3", "3", undefined, undefined]
 
-			} else if ( ( result = face_pattern3.exec( line ) ) !== null ) {
+					state.addFace(
+						result[ 1 ], result[ 3 ], result[ 5 ], result[ 7 ],
+						undefined, undefined, undefined, undefined,
+						result[ 2 ], result[ 4 ], result[ 6 ], result[ 8 ]
+					);
 
-				// ["f 1/1/1 2/2/2 3/3/3", " 1/1/1", "1", "1", "1", " 2/2/2", "2", "2", "2", " 3/3/3", "3", "3", "3", undefined, undefined, undefined, undefined]
+				} else if ( ( result = this.regexp.face_vertex.exec( line ) ) !== null ) {
 
-				addFace(
-					result[ 2 ], result[ 6 ], result[ 10 ], result[ 14 ],
-					result[ 3 ], result[ 7 ], result[ 11 ], result[ 15 ],
-					result[ 4 ], result[ 8 ], result[ 12 ], result[ 16 ]
-				);
+					// f vertex vertex vertex
+					// 0            1    2    3   4
+					// ["f 1 2 3", "1", "2", "3", undefined]
 
-			} else if ( ( result = face_pattern4.exec( line ) ) !== null ) {
+					state.addFace(
+						result[ 1 ], result[ 2 ], result[ 3 ], result[ 4 ]
+					);
 
-				// ["f 1//1 2//2 3//3", " 1//1", "1", "1", " 2//2", "2", "2", " 3//3", "3", "3", undefined, undefined, undefined]
+				} else {
 
-				addFace(
-					result[ 2 ], result[ 5 ], result[ 8 ], result[ 11 ],
-					undefined, undefined, undefined, undefined,
-					result[ 3 ], result[ 6 ], result[ 9 ], result[ 12 ]
-				);
+					throw new Error( "Unexpected face line: '" + line  + "'");
 
-			} else if ( ( result = object_pattern.exec( line ) ) !== null ) {
+				}
 
-				// o object_name
-				// or
-				// g group_name
+			} else if ( lineFirstChar === "l" ) {
 
-				var name = result[ 0 ].substr( 1 ).trim();
+				var lineParts = line.substring(1).trim().split(" ");
+				var lineVertexes = [], lineUVs = [];
 
-				if ( foundObjects === false ) {
+				if (line.indexOf("/") === -1) {
 
-					foundObjects = true;
-					object.name = name;
+					lineVertexes = lineParts;
 
 				} else {
+					for (var li = 0, llen = lineParts.length; li < llen; li++) {
 
-					addObject( name );
+						var parts = lineParts[li].split("/");
+						if (parts[0] !== "")
+							lineVertexes.push(parts[0]);
+						if (parts[1] !== "")
+							lineUVs.push(parts[1])
 
+					}
 				}
+				state.addLineGeometry(lineVertexes, lineUVs);
+
+			} else if ( ( result = this.regexp.object_pattern.exec( line ) ) !== null ) {
+
+				// o object_name
+				// or
+				// g group_name
+
+				var name = result[ 0 ].substr( 1 ).trim();
```

Comment:
This may be related to the comment I made above, but if we make the whitespace mandatory, then we can remove this substr I believe.